test/neural_networks/test_effective_dimension.py

```diff
@@ -0,0 +1,260 @@
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+""" Unit Tests for Effective Dimension Algorithm """
+
+import unittest
+
+from test import QiskitMachineLearningTestCase
+
+import numpy as np
+from ddt import ddt, data, unpack
+
+from qiskit import Aer, QuantumCircuit
+from qiskit.circuit.library import ZFeatureMap, RealAmplitudes
+from qiskit.utils import QuantumInstance, algorithm_globals
+from qiskit.utils import optionals
+
+from qiskit.opflow import PauliSumOp
+from qiskit_machine_learning.neural_networks import TwoLayerQNN, CircuitQNN
+from qiskit_machine_learning.neural_networks import (
+    EffectiveDimension,
+    LocalEffectiveDimension,
+)
+from qiskit_machine_learning import QiskitMachineLearningError
+
+
+@ddt
+class TestEffectiveDimension(QiskitMachineLearningTestCase):
+    """Test the Effective Dimension algorithm"""
+
+    @unittest.skipUnless(optionals.HAS_AER, "qiskit-aer is required to run this test")
+    def setUp(self):
+        super().setUp()
+
+        algorithm_globals.random_seed = 1234
+        qi_sv = QuantumInstance(
+            Aer.get_backend("aer_simulator_statevector"),
+            seed_simulator=algorithm_globals.random_seed,
+            seed_transpiler=algorithm_globals.random_seed,
+        )
+
+        # set up quantum neural networks
+        num_qubits = 3
+        feature_map = ZFeatureMap(feature_dimension=num_qubits, reps=1)
+        ansatz = RealAmplitudes(num_qubits, reps=1)
+
+        # CircuitQNNs
+        qc = QuantumCircuit(num_qubits)
+        qc.append(feature_map, range(num_qubits))
+        qc.append(ansatz, range(num_qubits))
+
+        def parity(x):
+            return f"{x:b}".count("1") % 2
+
+        circuit_qnn_1 = CircuitQNN(
+            qc,
+            input_params=feature_map.parameters,
+            weight_params=ansatz.parameters,
+            interpret=parity,
+            output_shape=2,
+            sparse=False,
+            quantum_instance=qi_sv,
+        )
+
+        # qnn2 for checking result without parity
+        circuit_qnn_2 = CircuitQNN(
+            qc,
+            input_params=feature_map.parameters,
+            weight_params=ansatz.parameters,
+            sparse=False,
+            quantum_instance=qi_sv,
+        )
+
+        # OpflowQNN
+        observable = PauliSumOp.from_list([("Z" * num_qubits, 1)])
+        opflow_qnn = TwoLayerQNN(
+            num_qubits,
+            feature_map=feature_map,
+            ansatz=ansatz,
+            observable=observable,
+            quantum_instance=qi_sv,
+        )
+
+        self.qnns = {"circuit1": circuit_qnn_1, "circuit2": circuit_qnn_2, "opflow": opflow_qnn}
+
+        # define sample numbers
+        self.n_list = [5000, 8000, 10000, 40000, 60000, 100000, 150000, 200000, 500000, 1000000]
+        self.n = 5000
+
+    @data(
+        # qnn_name, num_inputs, num_params, result
```

Comment:
```suggestion
        # qnn_name, num_inputs, num_weights, result
```
I think thats more correct than `params` now right. Its fairly minor being a comment in test case but may as well make it better for future readers